Web3Swift/StringScalar/CompactHexString.swift

```diff
@@ -0,0 +1,62 @@
+//
+// This source file is part of the Web3Swift.io open source project
+// Copyright 2018 The Web3Swift Authors
+// Licensed under Apache License v2.0
+//
+// CompactHexString.swift
+//
+// Created by Vadim Koleoshkin on 16/06/2018
+//
+
+import Foundation
+
+/** A string that represents hexadecimal numbers in a compact way */
+public final class CompactHexString: StringScalar {
+    
+    private let hex: StringScalar
+    
+    /**
+     Ctor
+     
+     - parameters:
+        - hex: a string describing a hexadecimal
+     */
+    public init(hex: StringScalar) {
+        self.hex = UnprefixedHexString(
```

Comment:
Use `TrimmedPrefixString` here instead of duplicating its functionality in the `value()` method

Reply:
It does not duplicate it. `UnprefixedHexString` == `TrimmedPrefixString` with `HexPrefix` argument.

Reply:
@rockfridrich what I mean is you can do it this way:

```swift
class CompactHexString: StringScalar {

    private let hex: StringScalar

    init(hex: StringScalar) {
        self.hex = TrimmedPrefixString(
            string: UnprefixedHexString(
                hex: hex
            ),
            prefix: SimpleString(
                string: "0"
            )
        )
    }

    func value() throws -> String {
        return try hex.string()
    }

}
```